src/CodeGen_LLVM.cpp

```diff
@@ -1614,6 +1614,12 @@ void CodeGen_LLVM::visit(const Load *op) {
 
     bool possibly_misaligned = (might_be_misaligned.find(op->name) != might_be_misaligned.end());
 
+    // If it's a Handle, load it as a uint64_t and then cast
+    if (op->type.is_handle()) {
```

Comment:
Ditto on asserting this is scalar. Will this always happen when loading e.g. the host pointer from a buffer_t? Any intuition on the performance impact if any?